### PR TITLE
Add option to pass custom derived data path to clear_derived_data

### PR DESCRIPTION
**Pull request by joshrlesch**
For our build process we sent the derived data to a custom folder and then remove before the next run. Right now we just call a `rm -rf` on that directory, but would like to use fastlane's method :)

### CODE REVIEW

File: lib/fastlane/actions/clear_derived_data.rb [hunk 1]
```diff
@@ -16,11 +22,16 @@ def self.description
...
-        "Deletes the Derived Data from '~/Library/Developer/Xcode/DerivedData'"
+        "Deletes the Derived Data from '~/Library/Developer/Xcode/DerivedData' or a supplied path"
       end
 
       def self.available_options
-        []
+        [
+          FastlaneCore::ConfigItem.new(key: :derived_data_path,
+                                       env_name: "DERIVED_DATA_PATH",
+                                       description: "Custom path for derivedData",
```
Comment (KrauseFx) [inline lib/fastlane/actions/clear_derived_data.rb]: This should have `default_value: "~/Library/Developer/Xcode/DerivedData"` and then expand_path at a later point

File: lib/fastlane/actions/clear_derived_data.rb [hunk 2]
```diff
@@ -16,11 +22,16 @@ def self.description
...
+                                       default_value: nil, optional: true)
```
Comment (KrauseFx) [inline lib/fastlane/actions/clear_derived_data.rb]: Please have one key/value per line only.

File: lib/fastlane/actions/clear_derived_data.rb [hunk 3]
```diff
@@ -1,8 +1,14 @@
 module Fastlane
   module Actions
     class ClearDerivedDataAction < Action
-      def self.run(params)
-        path = File.expand_path("~/Library/Developer/Xcode/DerivedData")
+      def self.run(options)
+        if options[:derived_data_path]
```
Comment (KrauseFx) [inline lib/fastlane/actions/clear_derived_data.rb]: You can revert these changes once you provide a `default_value` in the `options` array. Make sure to still run `File.expand_path(options[: derived_data_path])` when accessing the actual value.

**Comment by KrauseFx** [comment]
Looks great, thanks for the pull request, I added some comments before it's ready to be merged :+1:

File: lib/fastlane/actions/clear_derived_data.rb (hunk 1 above)
Comment (joshrlesch) [inline lib/fastlane/actions/clear_derived_data.rb]: done

**Comment by KrauseFx** [comment]
That's great, thanks for updating the pull request :+1:
